pyspectrafuse/commands/build_cluster_db.py

```
@@ -0,0 +1,390 @@
+                    COALESCE(s.precursor_mz, 0.0) AS precursor_mz,
+                    s.posterior_error_probability,
+                    s.global_qvalue,
+                    '{species}' AS species,
+                    '{instrument}' AS instrument
+                FROM read_parquet('{membership_tmp}') m
+                LEFT JOIN ({source_union}) s
+                    ON m.run_file_name = s.run_file_name
+                    AND m.orig_scan = s.orig_scan
+                ORDER BY m.cluster_id
+            ) TO '{psm_path}' (FORMAT PARQUET, COMPRESSION ZSTD)
+        """)
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)